p4include/arch.p4

```diff
@@ -0,0 +1,40 @@
+#ifndef __INCLUDE_ARCH_P4__
+#define __INCLUDE_ARCH_P4__
+
+#ifdef __TARGET_BMV2__
+ #ifdef __ARCH_V1MODEL__
+  #include "v1model.p4"
+ #endif
+ #ifdef __ARCH_PSA__
+  #include "psa.p4"
+ #endif
+#endif
+
+#ifdef __TARGET_FRONTEND__
+ #ifdef __ARCH_V1MODEL__
+  #include "v1model.p4"
+ #endif
+ #ifdef __ARCH_PSA__
+  #include "psa.p4"
+ #endif
+#endif
+
+#ifdef __TARGET_GRAPH__
+ #ifdef __ARCH_V1MODEL__
+  #include "v1model.p4"
+ #endif
+ #ifdef __ARCH_PSA__
+  #include "psa.p4"
+ #endif
+#endif
+
+#ifdef __TARGET_EBPF__
```

Comment:
This does not work: you cannot list all possible back-end here, because you don't know what they are. Where is the Tofino back-end, or the p4c-xdp back-ends, which are loaded as extensions? This breaks the modularity of the compiler. So far there no one piece that has to know about all back-ends; when you add a back-end it just appears. 

If you really want to do this you should do it as part of the build process, and this file should be automatically generated, in the same way makefiles are.

Reply:
Since the includes seem purely based on the `__ARCH_` symbols, is it even necessary to check `__TARGET_`anything?